appholder/src/main/java/com/android/mdl/app/util/CredentialUtil.kt

```diff
@@ -69,20 +68,30 @@ class CredentialUtil private constructor(
             SecureAreaImplementationState.Android -> createAndroidKeystoreSettings(
                 provisioningChallenge = CHALLENGE,
                 userAuthenticationRequired = provisionInfo.userAuthentication,
-                authTimeoutSeconds = provisionInfo.userAuthenticationTimeoutSeconds.toLong()
+                authTimeoutMillis = provisionInfo.userAuthenticationTimeoutSeconds * 1000L,
+                userAuthenticationType = provisionInfo.userAuthType(),
+                useStrongBox = provisionInfo.useStrongBox,
+                ecCurve = provisionInfo.androidAuthKeyCurveOption.toEcCurve(),
+                validUntil = provisionInfo.validityInDays.toTimestampFromNow()
             )
 
             SecureAreaImplementationState.BouncyCastle -> createBouncyCastleKeystoreSettings(
                 passphrase = provisionInfo.passphrase
             )
         }
         if (provisionInfo.userAuthentication) {
-            createAndroidKey(provisionInfo.userAuthenticationTimeoutSeconds.toLong())
+            createAndroidKey(
+                provisionInfo.userAuthenticationTimeoutSeconds * 1000L,
+                provisionInfo.userAuthType(),
+                provisionInfo.androidAuthKeyCurveOption.toEcCurve(),
+                provisionInfo.useStrongBox,
+                provisionInfo.validityInDays.toTimestampFromNow()
+            )
         } else if (provisionInfo.passphrase != null) {
             createBouncyCastleKey(provisionInfo.passphrase)
         }
```

Comment:
Delete `createAndroidKey()` and `createBouncyCastleKey()` and just call `createKey()` directly on the `SecureArea` object. In general, please try to reduce the number of places where there's SecureArea-specific codepaths - this is so it'll be easier to add support for a new `SecureArea` in the future.